test/e2e/user_cors_test.go

```diff
@@ -0,0 +1,105 @@
+package e2e
+
+import (
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/ghodss/yaml"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	operatorclient "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1"
+
+	test "github.com/openshift/cluster-kube-apiserver-operator/test/library"
+)
+
+var clusterDefaultCORSALlowedOrigins = []string{
+	`//127\.0\.0\.1(:|$)`,
+	`//localhost(:|$)`,
+}
+
+func TestAdditionalCORSAllowedOrigins(t *testing.T) {
+	// initialize clients
+	kubeConfig, err := test.NewClientConfigForTest()
+	require.NoError(t, err)
+	configClient, err := configclient.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+	operatorClient, err := operatorclient.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+	kubeAPIServerOperatorClient := operatorClient.KubeAPIServers()
+
+	// Check the cluster defaults
+	defaultConfig := getKubeAPIServerConfigOrFail(t, kubeAPIServerOperatorClient)
+	assert.Equal(t, clusterDefaultCORSALlowedOrigins, defaultConfig)
+
+	t.Logf("Cluster default CORSAllowedOrigins: %v", defaultConfig)
+
+	testCases := []struct {
+		additionalCORS     []string
+		expectedConfigCORS []string
+	}{
+		{
+			additionalCORS: []string{"//valid.domain.com(:|$)"},
+			expectedConfigCORS: []string{
+				`//127\.0\.0\.1(:|$)`,
+				`//localhost(:|$)`,
+				`//valid.domain.com(:|$)`,
+			},
+		},
+		{
+			additionalCORS: []string{"//something.*.now(:|$)", "//domain.foreign.it(:|$)"},
+			expectedConfigCORS: []string{
+				`//domain.foreign.it(:|$)`,
+				`//something.*.now(:|$)`,
+				`//127\.0\.0\.1(:|$)`,
+				`//localhost(:|$)`,
+			},
+		},
+		{
+			expectedConfigCORS: []string{
+				`//127\.0\.0\.1(:|$)`,
+				`//localhost(:|$)`,
+			},
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Logf("running test %d", i)
+		updateAPIServerClusterConfigSpec(configClient, func(apiserver *configv1.APIServer) {
+			apiserver.Spec.AdditionalCORSAllowedOrigins = tc.additionalCORS
+		})
+
+		var currentCORS []string
+		err = wait.PollImmediate(time.Second, 30*time.Minute, func() (bool, error) {
```

Comment:
This was an unreasonable timeout caused by typo, although the test still receives `[]string(nil)` from the observed config, I have to find out why that is.

Reply:
for this kind of test timeouts there is `wait.ForeverTestTimeout`.